api/api/utils/image_proxy/__init__.py

```diff
@@ -120,9 +124,87 @@ def _tally_client_response_errors(tallies, month: str, domain: str, status: int)
         logger.warning("Redis connect failed, thumbnail HTTP errors not tallied.")
 
 
+# thmbfail == THuMBnail FAILures; this key path will exist for every thumbnail requested, so it needs to be space efficient
+FAILURE_CACHE_KEY_TEMPLATE = "thmbfail:{ident}"
+
+
+def _cache_repeated_failures(_get):
+    """
+    Wrap ``image_proxy.get`` to cache repeated upstream failures
+    and avoid re-requesting images likely to fail.
+
+    Do this by storing a count for each media identifier of requested thumbnails.
+    When the upstream request succeeds, increment the key. When it fails, decrement it.
```

Comment:
As I understand it, isn't this the opposite of what we do? We track the failure_count, and increment when it fails/decrement when it succeeds.

Reply:
Oops, yes! I originally implemented a slightly different approach and forgot to update this docstring. I'll push a commit to address this and the edits @zackkrida proposed.